tests/test_nblist.py

```diff
@@ -376,7 +380,148 @@ def test_nblist_max_interactions(block_size, tiles):
+        host_config = HostConfig(host_sys, host_conf, box, host_conf.shape[0])
+    else:
+        assert 0, "Invalid host name"
+
+    st = SingleTopology(mol_a, mol_b, core, forcefield)
+    host = setup_optimized_host(st, host_config)
+
+    lambda_grid = np.array([0.0])
+
+    initial_state = setup_initial_states(st, host, DEFAULT_TEMP, lambda_grid, seed=2024, min_cutoff=None)[0]
+
+    return st, host, host_name, initial_state
+
+
+def compute_mean_tile_density(
+    nblist, frame: NDArray, box: NDArray, cutoff: float, column_block: int = 32, row_block: int = 32
+) -> float:
+    assert column_block == 32
+    assert column_block == row_block
+    N = frame.shape[0]
+    box_diag = np.diagonal(box)
+    assert nblist.get_num_row_idxs() == N
+    ixn_list = nblist.get_nblist(frame, box, cutoff)
+    tile_densities = []
+    num_blocks = (N + row_block - 1) // row_block
+    ixns_per_tile = min(N, column_block) * min(N, row_block)
+
+    row_block_offset = 0
+    # Each ixn_block is the column atoms in the tile that interact with at least one
+    # row block atom
+    for i, ixn_block in enumerate(ixn_list):
+        row_coords = frame[row_block_offset : row_block_offset + row_block]
+        row_coords = np.expand_dims(row_coords, axis=0)
+        column_ixn_atoms = np.array(ixn_block)
+        column_block_offset = 0
+        for _ in range(num_blocks - i):
+            # Find the interactions that would make up a tile, if the tile is empty
+            # don't compute density.
+            tile_column_ixns = column_ixn_atoms[
+                (column_ixn_atoms >= column_block_offset) & (column_ixn_atoms < (column_block_offset + column_block))
+            ]
+            column_block_offset += column_block
+            if tile_column_ixns.size == 0:
+                continue
+            col_coords = np.expand_dims(frame[tile_column_ixns], axis=1)
+            deltas = image_coords(row_coords - col_coords, box_diag)
+            dij = np.linalg.norm(deltas, axis=-1)
+            ixns = np.sum(dij < cutoff)
+            tile_densities.append(ixns / ixns_per_tile)
+        row_block_offset += row_block
+    return float(np.mean([tile_densities]))
+
+
+@pytest.mark.memcheck
+@pytest.mark.parametrize("cutoff", [1.0, 1.2])
+@pytest.mark.parametrize("precision", [np.float32])
+def test_nblist_density_dhfr(cutoff, precision):
+    _, _, coords, box = setup_dhfr()
+
+    if precision == np.float32:
+        nblist = custom_ops.Neighborlist_f32(coords.shape[0])
+    else:
+        nblist = custom_ops.Neighborlist_f64(coords.shape[0])
+    unsorted_density = compute_mean_tile_density(nblist, coords, box, cutoff)
+    print(f"DHFR NBList Occupancy with Cutoff {cutoff}, no sort: {unsorted_density * 100.0:.3g}%")
+    perm = hilbert_sort(coords, box)
+    coords = coords[perm]
+    density = compute_mean_tile_density(nblist, coords, box, cutoff)
+    print(f"DHFR NBList Occupancy with Cutoff {cutoff}: {density * 100.0:.3g}%")
+    assert density > 0.10
```

Comment:
This is 15% for a cutoff of 0.9, @proteneer does that seem right? Offline you had said you expect 13%, but after fixing up the incorrect computation of the tile density didn't see 13%.